### PR TITLE
feat: add shortcuts ui experiment

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- add shortcuts ui experiment
- add new one first session UI
- add new "Shortcuts +" button when dismissed

TODO:

- [x] see why openai logo is not resolving - opened [PR](https://github.com/dailydotdev/daily-scraper/pull/590) - openai has bot detection, I'll just use static images for this placeholder element
- [x] see if reordering of feed heading is [needed from design](https://www.figma.com/design/nmfWPS7x3kzLUvYMkBx2kW/daily.dev---Dev-Mode?node-id=9910-29515&m=dev) - asked [here](https://dailydotdev.slack.com/archives/C06LW384AF6/p1717428461931389)
- [x] see if we want to cleanup `onboardingMostVisited` experiment? - not yet

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [x] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-352 #done

### Preview domain
https://as-352-shortcuts-ui.preview.app.daily.dev

### CODE REVIEW

**Comment by capJavert** [comment]
Some open points but implementation is there so opening for review.

File: packages/shared/src/lib/featureManagement.ts [hunk 1]
```diff
@@ -54,6 +55,7 @@ const feature = {
   searchGoogle: new Feature('search_google', false),
   onboardingFlip: new Feature('onboarding_flip', false),
   improvedSharedPostCard: new Feature('improved_shared_post_card', false),
+  shortcutsUI: new Feature('shortcuts_ui', ShortcutsUIExperiment.Control),
```
Comment (nensidosari) [inline packages/shared/src/lib/featureManagement.ts:58]: Added in growthbook https://app.growthbook.io/features/shortcuts_ui

File: packages/extension/src/newtab/ShortcutLinks.tsx [hunk 2]
```diff
@@ -187,6 +206,88 @@ export default function ShortcutLinks({
     isShortcutsV1,
   ]);
 
+  const onOptionsOpen = () => {
+    setShowOptions(true);
+
+    trackEvent({
+      event_name: AnalyticsEvent.OpenShortcutConfig,
+      target_type: TargetType.Shortcuts,
+    });
```
Comment (rebelchris) [inline packages/extension/src/newtab/ShortcutLinks.tsx:215]: Not blocking, but nice to have test case for it :)

File: packages/shared/src/components/filters/MyFeedHeading.tsx [hunk 3]
```diff
@@ -36,6 +42,8 @@ function MyFeedHeading({
   const isLaptop = useViewSize(ViewSize.Laptop);
   const feedName = getFeedName(router.pathname);
   const { isCustomFeed } = useFeedName({ feedName });
+  const shortcutsUIFeature = useFeature(feature.shortcutsUI);
+  const isShortcutsUIV1 = shortcutsUIFeature === ShortcutsUIExperiment.V1;
```
Comment (rebelchris) [inline packages/shared/src/components/filters/MyFeedHeading.tsx:46]: This will do enrollment for everyone even if they not on extension even right?

**Review by rebelchris — COMMENTED** [review]
Just one blocking question around enrollment for this one.

File: packages/shared/src/components/filters/MyFeedHeading.tsx (hunk 3 above)
Comment (capJavert) [reply packages/shared/src/components/filters/MyFeedHeading.tsx:46]: yes, good catch

File: packages/extension/src/newtab/ShortcutLinks.tsx (hunk 2 above)
Comment (capJavert) [reply packages/extension/src/newtab/ShortcutLinks.tsx:215]: It is the old code I just moved it up, but I'll see if I can add test yes 👌

File: packages/shared/src/components/filters/MyFeedHeading.tsx (hunk 3 above)
Comment (capJavert) [reply packages/shared/src/components/filters/MyFeedHeading.tsx]: 3633f571c80d81f2fbb51f1c7fc67e171ff0b0fc

**Review by rebelchris — APPROVED** [review]
Nice this should do!

File: packages/extension/src/newtab/ShortcutLinks.tsx (hunk 2 above)
Comment (capJavert) [reply packages/extension/src/newtab/ShortcutLinks.tsx:215]: We have the test for this already

File: packages/shared/src/lib/featureManagement.ts (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/lib/featureManagement.ts:59]: thank you!

**Review by nensidosari — APPROVED** [review]

**Review by omBratteng — APPROVED** [review]

**Pull request merged**